src/libcore/iter/adapters/mod.rs

```diff
@@ -1609,6 +1609,100 @@ impl<I: Iterator, P> Iterator for TakeWhile<I, P>
     }
 }
 
+/// An iterator that only accepts elements while `predicate` returns `Some(_)`.
+///
+/// This `struct` is created by the [`take_while_map`] method on [`Iterator`]. See its
+/// documentation for more.
+///
+/// [`take_while_map`]: trait.Iterator.html#method.take_while
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "iter_take_while_map", reason = "recently added", issue = "0")]
+#[derive(Clone)]
+pub struct TakeWhileMap<I, P> {
+    iter: I,
+    flag: bool,
```

Comment:
Maybe, not just flag but `finished`?

Reply:
This field was copy-pasted from `TakeWhile` 🤷‍♀️ But I agree that `finished` is more clear than `flag` (actually when I was first reading `TakeWhile` sources I took me a while to understand the meaning of the `flag`)